File: Phyton/Gestor_canciones/src/gestorAplicacion/personas/Artista.py
from gestorAplicacion.personas.Persona import Persona
class Artista(Persona):
    id_artista=1
    artistas=[]
    def __init__(self, nombre, canciones, presentacion, genero):
        self.nombre=nombre
        self._canciones=canciones
        self._presentacion=presentacion
        self._genero=genero
        self._id_realA=Artista.id_artista
        Artista.id_artista+=1
        Artista.artistas.append(self)
    def __init__(self, nombre, genero):
        self.nombre=nombre
        self._canciones=[]
        self._presentacion="Falta descripcion..."
        self._genero=genero
        self._id_realA=Artista.id_artista
        Artista.id_artista+=1
        Artista.artistas.append(self)
    def __init__(self, nombre, presentacion, genero):
        self.nombre=nombre
        self._canciones=[]
        self._presentacion=presentacion
        self._genero=genero
        self._id_realA=Artista.id_artista
        Artista.id_artista+=1
        Artista.artistas.append(self)
    def datosPersona(self):
        return "Nombre: " + self.nombre + "\n" + "Tipo: Artista" + "\n" +"Presentacion en app: " + self._presentacion + "\n" + "número de canciones creadas: " + len(self._canciones)
    def getCanciones(self):
        return self._canciones
    def setCanciones(self,canciones):
        self._canciones=canciones
    def getPresentacion(self):
        return self._presentacion
    def setPresentacion(self, presentacion):
        self._presentacion=presentacion
    def getGenero(self):
        return self._genero
    def setGenero(self, genero):
        self._genero=genero
    def getId_artista(self):
        return self._id_realA

File: Gestor_canciones2/src/gestorAplicacion/personas/Artista.java
package gestorAplicacion.personas;

import java.util.ArrayList;
import java.io.Serializable;
import gestorAplicacion.elementosLibreria.*;


public class Artista implements Serializable, Persona {

	private static final long serialVersionUID = 1L;
	private static ArrayList<Artista> artistas;
	static {
		artistas = new ArrayList<Artista>();
	}
	
    public String nombre;
    private ArrayList<Cancion> canciones;
    private String presentacion;
    private Genero genero;
    private static int id_artista = 1;
	private int id_realA;
    
    
    //Creo que sobran atributos en ester constructor
	public Artista(String nombre, ArrayList<Cancion> canciones, String presentacion, Genero genero) {
		super();
		this.nombre = nombre;
		this.canciones = canciones;
		this.presentacion = presentacion;
		this.genero = genero;
		id_realA = id_artista;
		id_artista++;
		artistas.add(this);
	}
	
	public Artista(String nombre, Genero genero) {
		this.nombre = nombre;
		canciones = new ArrayList<Cancion>();
		presentacion = "Falta descripción...";
		this.genero = genero;
		id_realA = id_artista;
		id_artista++;
		artistas.add(this);
	}
	
	public Artista(String nombre, String presentacion, Genero genero) {
		this.nombre = nombre;
		canciones = new ArrayList<Cancion>();
		this.presentacion = presentacion;
		this.genero = genero;
		id_realA = id_artista;
		id_artista++;
		artistas.add(this);
	}
	
	public Artista(String nombre, ArrayList<Cancion> canciones, Genero genero) {
		this.nombre = nombre;
		this.canciones = canciones;
		presentacion = "Falta descripción...";
		this.genero = genero;
		id_realA = id_artista;
		id_artista++;
		artistas.add(this);
	}
	public String datosPersona() {
		return 
				"Nombre: " + nombre + "\n" + 
				"Tipo: Artista" + "\n" +
				"Presentacion en app: " + presentacion + "\n" + 
				"número de canciones creadas: " + (canciones.size());
	}
	public static ArrayList<Artista> getArtistas() {
		return artistas;
	}

	public static void setArtistas(ArrayList<Artista> artistas) {
		Artista.artistas = artistas;
	}

	public String getNombre() {
		return nombre;
	}

	public void setNombre(String nombre) {
		this.nombre = nombre;
	}

	public void setPresentacion(String presentacion) {
		this.presentacion = presentacion;
	}

	public ArrayList<Cancion> getCanciones() {
		return canciones;
	}

	public void setCanciones(ArrayList<Cancion> canciones) {
		this.canciones = canciones;
	}

	public String getPresentacion() {
		return presentacion;
	}

	public void change_Presentacion(String nuevo) {
		this.presentacion = nuevo;
	}

	public Genero getGenero() {
		return genero;
	}

	public void setGenero(Genero genero) {
		this.genero = genero;
	}

	public int getId_artista() {
		return id_realA;
	}

	public void setId_artista(int id_artista) {
		this.id_realA = id_artista;
	}
	
	public void agg_Cancion(Cancion cancion) {
		canciones.add(cancion);
	}
	public String elim_Cancion(Cancion cancion) {
    	boolean presente=false;
    	for (Cancion c: canciones) {
    		if (cancion==c){
    			presente=true; 
    		}
    
    	}
    	if (presente==true) {
    		canciones.remove(cancion);
    		return"cancion eliminada";
    	}
    	else {
    		return"ERROR: la cancion no esta subida o no es del artista";
    	}
    }

    
    
}
